service/src/test/java/bio/terra/tanagra/service/FeatureSetServiceTest.java

```diff
@@ -246,6 +246,123 @@ void listAllOrSelected() {
+                    ResourceType.FEATURE_SET, ResourceId.forStudy(study1.getId())),
+                0,
+                10)
+            .size());
+    assertEquals(featureSet1.getUnderlay(), clonedFeatureSet1.getUnderlay());
+    assertEquals("Copy of: " + featureSet1.getDisplayName(), clonedFeatureSet1.getDisplayName());
+    assertEquals(newDescription, clonedFeatureSet1.getDescription());
+    assertEquals(USER_EMAIL_2, clonedFeatureSet1.getCreatedBy());
+    assertEquals(USER_EMAIL_2, clonedFeatureSet1.getLastModifiedBy());
+    assertEquals(clonedFeatureSet1.getCreated(), clonedFeatureSet1.getLastModified());
+    assertTrue(CollectionUtils.isEmpty(clonedFeatureSet1.getCriteria()));
+    assertTrue(CollectionUtils.isEmpty(clonedFeatureSet1.getExcludeOutputAttributesPerEntity()));
+
+    // Create second feature set with criteria and attributes
+    List<Criteria> criteria2 = List.of(GENDER_EQ_WOMAN.getValue());
+    Map<String, List<String>> attributes2 = Map.of(GENDER_EQ_WOMAN.getKey(), PERSON_ATTRIBUTES);
+    FeatureSet featureSet2 =
+        featureSetService.createFeatureSet(
+            study1.getId(),
+            FeatureSet.builder()
+                .underlay(UNDERLAY_NAME)
+                .displayName("feature set 2")
+                .description("second feature set")
+                .criteria(criteria2)
+                .excludeOutputAttributesPerEntity(attributes2),
+            USER_EMAIL_2);
+    assertNotNull(featureSet2);
+    LOGGER.info("Created feature set {} at {}", featureSet2.getId(), featureSet2.getCreated());
+
+    // Clone the feature set without new description and verify
+    String newDisplayName = "cloned feature set displayName";
+    FeatureSet clonedFeatureSet2 =
+        featureSetService.cloneFeatureSet(
+            study1.getId(), featureSet2.getId(), USER_EMAIL_2, newDisplayName, null);
+    assertEquals(
+        4,
+        featureSetService
+            .listFeatureSets(
+                ResourceCollection.allResourcesAllPermissions(
+                    ResourceType.FEATURE_SET, ResourceId.forStudy(study1.getId())),
+                0,
+                10)
+            .size());
+    assertEquals(newDisplayName, clonedFeatureSet2.getDisplayName());
+    assertEquals("Copy of: " + featureSet2.getDescription(), clonedFeatureSet2.getDescription());
+
+    List<Criteria> clonedCriteria2 = clonedFeatureSet2.getCriteria();
+    assertEquals(criteria2.size(), clonedCriteria2.size());
+    for (int i = 0; i < criteria2.size(); ++i) {
+      assertEquals(criteria2.get(i), clonedCriteria2.get(i));
+    }
+
+    Map<String, List<String>> clonedAttributes2 =
+        clonedFeatureSet2.getExcludeOutputAttributesPerEntity();
+    assertEquals(attributes2.size(), clonedAttributes2.size());
+    attributes2.forEach(
+        (key, value) ->
+            assertEquals(
+                value.stream().sorted().toList(), clonedAttributes2.get(key).stream().toList()));
```

Comment:
Do you need the `sorted()` here?

Reply:
Yes - I tried it without, and the test failed --https://github.com/DataBiosphere/tanagra/actions/runs/10725173346/job/29742423554#step:8:239